### PR TITLE
bug: [418692] Optional tag missing for single component when set to optional

### PR DESCRIPTION
**Pull request by KiranVarma02**
https://dev.azure.com/defragovuk/DEFRA-CDP/_boards/board/t/DXT%20Team/Stories?System.IterationPath=%40currentIteration&workitem=428692

### CODE REVIEW

File: src/server/plugins/engine/pageControllers/PageControllerBase.ts [hunk 1]
```diff
@@ -146,8 +146,13 @@ export class PageControllerBase {
...
           labelOrLegend.isPageHeading = isPageHeading
 
+          // Check for optional in label
+          const isOptional = labelOrLegend.text.includes('optional')
+
           if (pageTitle) {
-            labelOrLegend.text = pageTitle
+            labelOrLegend.text = isOptional
+              ? pageTitle + ' (optional)'
+              : pageTitle
```
Comment (KiranVarma02) [inline src/server/plugins/engine/pageControllers/PageControllerBase.ts:155]: I am not convinced this is best solution but I didn't find any other way to do it. 

@alexluckett , @davidjamesstone  - `heading.html` is not the file used to render from my investigation. govukInput in `textfield.html` renders label, so I made above change. 

This solution works. 

Please let me know if I missed anything or better solution.

Comment (KiranVarma02) [reply src/server/plugins/engine/pageControllers/PageControllerBase.ts:155]: Demo'd and tested few components with Naz. All looks good.

@alexluckett

File: src/server/plugins/engine/pageControllers/PageControllerBase.ts [hunk 2]
```diff
@@ -146,8 +146,14 @@ export class PageControllerBase {
         if (isPageHeading) {
           labelOrLegend.isPageHeading = isPageHeading
 
+          // Check for optional in label
+          const isOptional =
+            this.components.formItems.at(0)?.options.required === false
```
Comment (colinrotherham) [inline src/server/plugins/engine/pageControllers/PageControllerBase.ts:151]: Is this the same as `formComponents[0]` above?

Sorry not checked properly! Thanks @KiranVarma02

```suggestion
            formComponents[0].options.required === false
```

File: src/server/plugins/engine/pageControllers/PageControllerBase.ts [hunk 3]
```diff
@@ -146,8 +146,14 @@ export class PageControllerBase {
...
+
           if (pageTitle) {
-            labelOrLegend.text = pageTitle
+            labelOrLegend.text = isOptional
+              ? pageTitle + ' (optional)'
```
Comment (colinrotherham) [inline src/server/plugins/engine/pageControllers/PageControllerBase.ts:155]: Mind using the export from **src/server/plugins/engine/components/constants.ts**?

```suggestion
              ? `${pageTitle}${optionalText}`
```

File: src/server/plugins/engine/pageControllers/PageControllerBase.ts (hunk 2 above)
Comment (KiranVarma02) [reply src/server/plugins/engine/pageControllers/PageControllerBase.ts:151]: No, its not same. formComponents doesn't have options data.

File: src/server/plugins/engine/pageControllers/PageControllerBase.ts (hunk 3 above)
Comment (KiranVarma02) [reply src/server/plugins/engine/pageControllers/PageControllerBase.ts]: done

File: test/condition/text.test.js [hunk 4]
```diff
@@ -33,7 +33,7 @@ describe('TextField based conditions', () => {
     const { container } = await renderResponse(server, options)
 
     const $input = container.getByRole('textbox', {
-      name: 'First page'
+      name: 'First page (optional)'
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply test/condition/text.test.js:36]: done

**Review by alexluckett — APPROVED** [review]

**Pull request merged**